junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/RunnerTestDescriptorPostProcessor.java

```diff
@@ -35,18 +35,17 @@
 
 	private final UniqueIdReader uniqueIdReader = new UniqueIdReader();
 	private final UniqueIdStringifier uniqueIdStringifier = new UniqueIdStringifier();
-	private final TestSourceProvider testSourceProvider;
-
-	public RunnerTestDescriptorPostProcessor(TestSourceProvider testSourceProvider) {
-		this.testSourceProvider = testSourceProvider;
-	}
+	private final TestSourceProvider testSourceProvider = new TestSourceProvider();
 
 	void applyFiltersAndCreateDescendants(RunnerTestDescriptor runnerTestDescriptor) {
 		addChildrenRecursively(runnerTestDescriptor);
 		runnerTestDescriptor.applyFilters(this::addChildrenRecursively);
 	}
 
 	private void addChildrenRecursively(VintageTestDescriptor parent) {
+		if (parent.getDescription().isTest()) {
```

Comment:
At a first glance, this looked a bit suspect, but I suppose that `VintageTestDescriptor#getType()` will never conceivably return `Type.CONTAINER_AND_TEST` (not even in a future release of Vintage) -- right?

Reply:
No, and `Description.isTest()` returns true iff it doesn't have children which is what the rest of this method is about.

Reply:
OK